File: alphatimer-protocol-api/src/main/java/org/lisasp/alphatimer/api/protocol/events/dropped/UnstructuredInputDroppedEvent.java
package org.lisasp.alphatimer.api.protocol.events.dropped;

import lombok.RequiredArgsConstructor;
import lombok.Value;
import org.lisasp.alphatimer.api.protocol.data.Bytes;

import java.time.LocalDateTime;

@Value
@RequiredArgsConstructor
public class UnstructuredInputDroppedEvent implements InputDataDroppedEvent {
    private final LocalDateTime timestamp;
    private final String competition;
    private final Bytes data;

    public UnstructuredInputDroppedEvent(LocalDateTime timestamp, String competition, byte[] data) {
        this(timestamp, competition, new Bytes(data));
    }
}

File: protocol-api/src/main/java/org/lisasp/alphatimer/api/protocol/events/dropped/UnstructuredInputDroppedEvent.kt
package org.lisasp.alphatimer.api.protocol.events.dropped

import org.lisasp.alphatimer.api.protocol.data.Bytes

data class UnstructuredInputDroppedEvent(private val data: Bytes) : InputDataDroppedEvent {
    constructor(data: ByteArray) : this(Bytes(data))
}